test/chain-specific-spokepools/Arbitrum_SpokePool.ts

```diff
@@ -23,14 +23,28 @@ describe("Arbitrum Spoke Pool", function () {
 
     l2GatewayRouter = await createFake("L2GatewayRouter");
 
-    arbitrumSpokePool = await (
-      await getContractFactory("Arbitrum_SpokePool", owner)
-    ).deploy(0, l2GatewayRouter.address, owner.address, hubPool.address, l2Weth, timer.address);
+    arbitrumSpokePool = await hre.upgrades.deployProxy(
+      await getContractFactory("Arbitrum_SpokePool", owner),
+      [0, l2GatewayRouter.address, owner.address, hubPool.address, l2Weth, timer.address],
+      { kind: "uups" }
+    );
 
     await seedContract(arbitrumSpokePool, relayer, [dai], weth, amountHeldByPool);
     await arbitrumSpokePool.connect(crossDomainAlias).whitelistToken(l2Dai, dai.address);
   });
 
+  it("Only cross domain owner upgrade logic contract", async function () {
+    // TODO: Could also use upgrades.prepareUpgrade but I'm unclear of differences
+    const implementation = await hre.upgrades.deployImplementation(
+      await getContractFactory("Arbitrum_SpokePool", owner),
+      { kind: "uups" }
+    );
+
+    // upgradeTo fails unless called by cross domain admin
+    await expect(arbitrumSpokePool.upgradeTo(implementation)).to.be.reverted;
```

Comment:
Is it worthwhile specifying the expected revert message for these? Then it's clear exactly where it's being caught.